smt/applications/ego.py

```diff
@@ -207,33 +226,38 @@ def _setup_optimizer(self, fun):
         ndarray: initial coord-y doe = fun(xdoe)
 
         """
-        # Set the bounds of the optimization problem
-        xlimits = self.options["xlimits"]
+        # Set the model
+        self.gpr = self.options["surrogate"]
+
+        # Set the continuous bounds of the optimization problem
+        self.xlimits = self.gpr._relax_limits(self.options["xlimits"])
 
         # Build initial DOE
-        self._sampling = LHS(xlimits=xlimits, criterion="ese")
+        self._sampling = LHS(xlimits=self.xlimits, criterion="ese")
         self._evaluator = self.options["evaluator"]
-
-        xdoe = self.options["xdoe"]
+        tunnel = self.options["tunnel"]
+        xdoe = self.options["xdoe"] 
         if xdoe is None:
             self.log("Build initial DOE with LHS")
             n_doe = self.options["n_doe"]
             x_doe = self._sampling(n_doe)
+
         else:
             self.log("Initial DOE given")
             x_doe = np.atleast_2d(xdoe)
 
+        x_doe = self.gpr._project_values(x_doe)
         ydoe = self.options["ydoe"]
         if ydoe is None:
-            y_doe = self._evaluator.run(fun, x_doe)
+            y_doe = self._evaluator.run(
+                fun, self.gpr._assign_labels(x_doe, self.options["xlimits"])
```

Comment:
_assign_labels should be renamed assign_labels. The underscore in front of a method name specify a protected method normally not used outside of the class or subclasses. same thing for project_values